client/src/models/storeMixins.ts

```diff
@@ -0,0 +1,196 @@
+import _ from "lodash";
+
+import { completeAssign } from "src/general_utils";
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+class StaticStore {
+  static register: (id: string, instance: StaticStore) => void;
+  static lookup: (id: string) => StaticStore | undefined;
+  static get_all: () => StaticStore[];
+  static __store__: Map<string, StaticStore>;
+}
+
+export const getStaticStore: () => typeof StaticStore = () => {
+  const _storeMap = new Map<string, StaticStore>();
+  StaticStore.register = (id: string, instance: StaticStore) => {
+    _storeMap.set(id, instance);
+  };
+  StaticStore.lookup = (id: string) => _storeMap.get(id);
+  StaticStore.get_all = () => _.uniq(Array.from(_storeMap.values()));
+  StaticStore.__store__ = _storeMap;
+  return StaticStore;
+};
```

Comment:
up until here is the only actual change. Below hasn't changed.

There is one major problem tho. `_storeMap` is storing all types of these static stores. So by the end, `_storeMap` has tables, glossary entries, footnotes, etc in that one single instance of variable.

Theoretically, I don't think this should be happening because `_storeMap` is being initialized as a new Map object every time this function is called. @Stephen-ONeil do you know why?

Trying different things on mixins to force it to work made me wonder the future of store mixins (and basically all of our models, subjects, etc) when we fully drop Table API and fully transition to server side. Will we keep the client data structures as is, but only swap out how it's coming in? Or refactor it to more appropriately suit the new data flow?

Reply:
Initializing a new map, but not a new `StaticStore`. Each call to `getStaticStore` mutates the static properties on the `StaticStore` class itself, clobbering over the map referenced by `__store__` each time. Everything extending from it is sharing this one static `__store__` (so there's a second bug here, which is that timing on calls to `getStaticStore` might lead to _wiping_ any stores that have been populated already).

I'm adopting the branch, so I'll sort that out :+1: